### PR TITLE
test: add a couple of tests for systemd-journal-{remote,upload}

### PR DESCRIPTION
**Pull request by mrc0mmand**
\+ a couple of accompanying fixes.

I got one question for @keszybz - since 8201af08fa09c2bd0f005fbe262f27e2c5bd2d86 (yeah, it's from 2014) when client certs are used, systemd-journal-remote uses DN (distinguished name) of the client certificate as a hostname. E.g. after connecting with systemd-journal-upload to systemd-journal-remote with a valid client certificate, we open a new journal file named `/var/log/journal/remote/remote-C=CZ,L=Brno,O=Foo,OU=Bar,CN=Test\x20Client.journal`:

```
systemd-journal-remote running as pid 781
Connection from C=CZ,L=Brno,O=Foo,OU=Bar,CN=Test Client
Certificate status: The certificate is trusted.
Fixed min_use=16.0M max_use=137.7M max_size=17.2M min_size=512.0K keep_free=68.8M n_max_files=100
Reserving 333 entries in field hash table.
Reserving 31345 entries in data hash table.
Journal effective settings seal=no keyed_hash=yes compress=ZSTD compress_threshold_bytes=512B
Opened output file /var/log/journal/remote/remote-C=CZ,L=Brno,O=Foo,OU=Bar,CN=Test\x20Client.journal
Creating source for passive fd:9 (C=CZ,L=Brno,O=Foo,OU=Bar,CN=Test Client)
```

Is that really expected? I guess I'm thrown off by the code calling it a *hostname*, as I'd imagine an IP address/FQDN of the other side there instead of cert's DN, but maybe I'm just overthinking this.

### CODE REVIEW

File: src/journal-remote/journal-remote.c [hunk 1]
```diff
@@ -30,6 +30,18 @@
 
 #define filename_escape(s) xescape((s), "/ ")
 
+#if HAVE_MICROHTTPD
+void MHDDaemonWrapper_free(MHDDaemonWrapper *d) {
+        if (!d)
+            return;
```
Comment (poettering) [inline src/journal-remote/journal-remote.c]: indentation should be 16spaces not 12

File: src/journal-remote/journal-remote.c [hunk 2]
```diff
@@ -30,6 +30,18 @@
...
+
+        MHD_stop_daemon(d->daemon);
```
Comment (poettering) [inline src/journal-remote/journal-remote.c]: the microhttp docs do not say what happens if you pass a NULL pointer to MHD_stop_daemon(). For MHDDaemonWrapper_free() to act as a destructor that is robust in regards to getting half-allocated objects passed in you need to add an `if (d->daemon)` first here hence.

File: src/journal-remote/journal-remote.c [hunk 3]
```diff
@@ -30,6 +30,18 @@
 
 #define filename_escape(s) xescape((s), "/ ")
 
+#if HAVE_MICROHTTPD
+void MHDDaemonWrapper_free(MHDDaemonWrapper *d) {
```
Comment (poettering) [inline src/journal-remote/journal-remote.c]: maybe if you touch this change this to be a proper destructor like we do it in our codebase these days, i.e. return NULL in the same type as you pass in, i.e. change to:

```c
MHDDaemonWrapper *MHDDaemonWrapper_fre(MHDDaemonWrapper *d) {
        if (!d)
                return NULL;

        …
        return mfree(d);
}
```

you get the idea

File: src/journal-remote/journal-remote.c [hunk 4]
```diff
@@ -353,6 +361,9 @@ int journal_remote_server_init(
 void journal_remote_server_destroy(RemoteServer *s) {
         size_t i;
 
+        if (!s)
+            return;
```
Comment (poettering) [inline src/journal-remote/journal-remote.c]: 16spaces

File: src/journal-remote/journal-remote-main.c [hunk 5]
```diff
@@ -841,31 +841,31 @@ static int parse_argv(int argc, char *argv[]) {
                 case ARG_URL:
                         if (arg_url)
                                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                       "cannot currently set more than one --url");
+                                                       "Cannot currently set more than one --url");
```
Comment (poettering) [inline src/journal-remote/journal-remote-main.c]: if you touch these, please also change `--url` to `--url=`, as we usually do when referencing cmdline options that take arguments in our docs and log messages. similar for the other log lines you are touching here.

File: src/journal-remote/journal-remote.c (hunk 1 above)
Comment (mrc0mmand) [reply src/journal-remote/journal-remote.c]: Whoops.

File: src/journal-remote/journal-remote.c (hunk 4 above)
Comment (mrc0mmand) [reply src/journal-remote/journal-remote.c]: D'oh, who's the little rascal that keeps eating my spaces.

**Review by bluca — APPROVED** [review]

**Pull request merged**